src/main/java/de/tum/in/www1/artemis/service/util/structureoraclegenerator/OracleGenerator.java

```diff
@@ -204,8 +203,10 @@ private static void walkProjectFileStructure(File file, List<File> foundFiles) {
 
         if (file.isDirectory()) {
             String[] subFiles = file.list();
-            for (String subFile : subFiles) {
-                walkProjectFileStructure(new File(file, subFile), foundFiles);
+            if (subFiles != null) {
```

Comment:
Good suggestions. To be honest, I did not implement the original code (which was really bad), but mostly refactored and improved since then. As this works now, I would also propose  to move this to another PR, so that we can merge this one now. Could you create an issue for this? Thanks!